### PR TITLE
Kapil august integration

### PR DESCRIPTION
**Pull request by kapil-ms**

### CODE REVIEW

File: DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs [hunk 1]
```diff
@@ -6,44 +6,73 @@
...
 
-        ~DeviceLogicTests() { }
+        [Fact]
+        public async void GetDevicesTest()
+        {
+            DeviceListQuery q = fixture.Create<DeviceListQuery>();
+            DeviceListQueryResult r = fixture.Create<DeviceListQueryResult>();
+            _deviceRegistryListRepositoryMock.SetupSequence(x => x.GetDeviceList(It.IsAny<DeviceListQuery>()))
+                .ReturnsAsync(r)
+                .ReturnsAsync(new DeviceListQueryResult());
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs]: what does 2 ReturnsAsync do ?

Comment (kapil-ms) [inline DeviceAdministration/Infrastructure.UnitTests/Infrastructure/DeviceLogicTests.cs]: I am using SetupSequence, so I can send different return values for recurring function calls
I can actually wanted to do a condition based returns, but lambda functions kept giving an error, so ended up using this.

File: DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs [hunk 2]
```diff
@@ -68,13 +69,14 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
                         true,
                         false) as IEnumerable;
 
-                deviceList = JsonConvert.DeserializeObject<List<DeviceModel>>(docs.ToString());
+                tmpDeviceList = JsonConvert.DeserializeObject<List<DeviceModel>>(docs.ToString());
+                deviceList.AddRange(tmpDeviceList);
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs]: what was the problem with initializing directly to deviceList?

Comment (kapil-ms) [inline DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs]: so, its a while loop running, we want to add more devices to deviceList in each iteration. If we initialize to deviceList, only the last iteration will be retained.

File: DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs [hunk 3]
```diff
@@ -68,13 +69,14 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
...
 
                 continuationToken = result.ContinuationToken;
 
             } while (!String.IsNullOrEmpty(continuationToken));
 
-            return deviceList;
+            return (deviceList.Count != 0 ? deviceList : null);
```
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs]: why not send an empty list ?

File: DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs (hunk 2 above)
Comment (handeadiguzel) [inline DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs]: oh okay it's in a loop

**Pull request merged**